### PR TITLE
Add multi-line command parsing

### PR DESCRIPTION
**Pull request by OCHyams**
Commands can be split over multiple lines provided that there is no extraneous
text (e.g. C++ code) between the command's parentheses.

### CODE REVIEW

File: tests/nostdlib/gcd/test.cpp [hunk 1]
```diff
@@ -17,5 +17,32 @@ int main()
 	return GCD(111, 259);
 }
```
Comment (OCHyams) [inline tests/nostdlib/gcd/test.cpp]: The indentation is pretty massive here but the file was already using tabs and not spaces.

File: dex/command/ParseCommand.py [hunk 2]
```diff
@@ -88,47 +88,79 @@ def _get_command_name(command_raw):
 
 def _find_all_commands_in_file(path, file_lines, valid_commands):
     commands = defaultdict(dict)
-
     err = CommandParseError()
     err.filename = path
-
+    parens = 0
     for lineno, line in enumerate(file_lines):
```
Comment (TomWeaver18) [inline dex/command/ParseCommand.py]: I think this implementation is cool, but I do have some misgivings on the style.

I'd like to see this function broken up into smaller chunks if possible (I don't mind bloat over readability).

The general control flow of the algo is hard to follow, I think this is due to over use of continues and breaks and the number of lines they're spread over.

**Comment by OCHyams** [comment]
I didn't see Tom's comment, I'll refactor it a bit now.

**Comment by OCHyams** [comment]
Refactored.

**Comment by TomWeaver18** [comment]
LGTM, thanks for taking my feedback comments onboard.

**Comment by OCHyams** [comment]
Rebased and squashed.

I don't do any checking for // style comment blocks because I think it would be very easy to generate false positives. The // could be embedded in a string, and then checking for the end of the string requires checking for the different formats of [string literals](https://en.cppreference.com/w/cpp/language/string_literal) as well as avoiding escaped quotes.


Reporting an error seems just as hard to me. But not impossible, I could give it a go?

**Comment by jmorse** [comment]
True, I was imagining people who used comments only on lines where there was no C/C++, I guess they could do more complicated things.

It's probably best to just document multi-line commands as only working in C-style comments /**/ in that case.
